backend/src/config.rs

```diff
@@ -1,16 +1,107 @@
-use cli::Config;
+use cli::{Config, LoggerConfig};
+use core::fmt;
+use error_stack::Context;
+use file_rotate::{suffix::AppendTimestamp, ContentLimit, FileRotate};
+use std::fmt::Display;
+use thiserror::Error;
 use tower_http::cors::CorsLayer;
+use tracing_appender::non_blocking::WorkerGuard;
+use tracing_subscriber::{filter::LevelFilter, prelude::*};
 
+#[allow(clippy::module_name_repetitions)]
 pub trait ConfigExt {
     /// Return either very permissive [`CORS`](`CorsLayer`) configuration
     /// or empty one based on `cors_allow_all` field
     fn get_cors_configuration(&self) -> CorsLayer;
 }
 
+pub trait LoggerExt {
+    /// Initialize logger.
+    ///
+    /// Returns [`WorkerGuard`]s for off-thread writers.
+    /// Should not be dropped.
+    ///
+    /// # Errors
+    ///
+    /// Function returns error if `init_file_rotate` fails
+    fn init_logger(&self) -> Result<Vec<WorkerGuard>, LoggerError>;
+
+    /// Returns [`std:io::Write`] object that rotates files on write
+    ///
+    /// # Errors
+    ///
+    /// Function returns error if `log_file` is not specified
+    fn init_file_rotate(&self) -> Result<FileRotate<AppendTimestamp>, LoggerError>;
+}
+
 impl ConfigExt for Config {
     fn get_cors_configuration(&self) -> CorsLayer {
         self.cors_allow_all
             .then_some(CorsLayer::very_permissive())
             .unwrap_or_default()
     }
 }
+
+impl LoggerExt for LoggerConfig {
+    fn init_logger(&self) -> Result<Vec<WorkerGuard>, LoggerError> {
+        let mut guards = Vec::with_capacity(2);
+        let file_writer = self.file.as_ref().map_or_else(
```

Comment:
You should not create an appender if it is disabled in config. `file_write` should be `Option<>` and have `None` value if logging to file is disabled